### PR TITLE
refactor: separate app router headless ssr examples

### PR DESCRIPTION
**Pull request by mrrajamanickam-coveo**
* In prep for adding a pages router example move existing app router src under a `app-router` dir
  - so as to minimize merge conflicts with app-router

TODO
- [x] Check if the e2e ssr samples tests are executing in CI as expected.

https://coveord.atlassian.net/browse/KIT-2795

### CODE REVIEW

File: packages/samples/headless-ssr/pages-router/.gitignore [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (mrrajamanickam-coveo) [inline packages/samples/headless-ssr/pages-router/.gitignore:1]: Registers the `pages-router` dir with a file as the dir is linked from readme.

File: packages/headless/src/app/commerce-engine/commerce-engine.ts [hunk 2]
```diff
@@ -22,7 +22,11 @@ import {cartReducer} from '../../features/commerce/context/cart/cart-slice';
 
 export type {CommerceEngineConfiguration};
 
-const commerceEngineReducers = {productListing: productListingV2Reducer, commerceContext: contextReducer, cart: cartReducer};
+const commerceEngineReducers = {
```
Comment (mrrajamanickam-coveo) [inline packages/headless/src/app/commerce-engine/commerce-engine.ts:25]: looks like bunch of commerce files (which probably were not lint staged) got included guessing due to lint staging of master merge commit

**Review by louis-bompart — APPROVED** [review]

**Review by Spuffynism — APPROVED** [review]

**Review by btaillon — APPROVED** [review]
good idea!

**Comment by mrrajamanickam-coveo** [comment]
Unrelated quantic test failure, proceeding to merge.

**Pull request merged**